include/onnxruntime/core/session/onnxruntime_session_options_config_keys.h

```diff
@@ -51,6 +51,9 @@ static const char* const kOrtSessionOptionsEnableQuantQDQ = "session.enable_quan
 // GeluApproximation has side effects which may change the inference results. It is disabled by default due to this.
 static const char* const kOrtSessionOptionsEnableGeluApproximation = "optimization.enable_gelu_approximation";
 
+// Enable or disable using arena for allocating initialized tensor memory. "1": disable; "0": enable. The default is "0".
+static const char* const kOrtSessionOptionsDisableArenaForInitializedTensorMemory = "optimization.disable_arena_for_initialized_tensor_memory";
```

Comment:
But isn't `ReserveMemoryForInitializers` not indicative of what is happening ? Aren't we always going to reserve memory for initializers. From where (arena vs non-arena) seems to be the differentiator. Thoughts ?

Reply:
Unfortunately, since the reservation is done by the arena allocator itself, this condition is confusing to read ```if (disable_arena_for_initialized_tensor_memory_allocation && default_cpu_alloc->Info().alloc_type == OrtArenaAllocator)```.
How about UseDeviceAllocatorForInitializers which says even if you're using an arena you ought to be using device allocators to allocate memory for initializers (now whether this is implemented using a separate allocator for initializers or as part of the arena allocator is an implementation detail)?

Reply:
sure, fixed

Reply:
Not sure that many people would have an idea of what 'using device allocator' means. Another suggestion would be calling this something like 'allocate initializers directly'. 

Alternatively the comment needs to be expanded to clarify what difference 'using device allocator' makes.

Reply:
Added a comment